# In this notebook I clean the data from the three lyric datasets.

i'll start by importing the required packages...

In [9]:
import pandas as pd
import numpy as np

# Folk Lyrics

I'll start by reading in the csv.

In [10]:
folk_df = pd.read_csv('/Users/liamgentile/Desktop/lyric_files/folk_lyrics.csv', converters={'lyrics': eval})

In [11]:
# display the first 5 rows
folk_df.head(5)

Unnamed: 0                                             lyrics  \
0           0  [Living on free food tickets. Water in the mil...   
1           1  [Do you remember days not so very long ago, wh...   
2           2  [Yes, I been dreaming about you every day, eac...   
3           3  [I'll sing you a song of Spiro Agnew and all t...   
4           4  [If I look like home to you, if I am your sign...   

                    song      artist  
0  theloveofcommonpeople  johndenver  
1  catchanotherbutterfly  johndenver  
2               daydream  johndenver  
3  theballadofspiroagnew  johndenver  
4                 circus  johndenver

We have an additional, unnecessary index, so I'll drop that using df.drop()

In [12]:
folk_df.drop('Unnamed: 0', axis=1, inplace=True)
folk_df

lyrics  \
0     [Living on free food tickets. Water in the mil...   
1     [Do you remember days not so very long ago, wh...   
2     [Yes, I been dreaming about you every day, eac...   
3     [I'll sing you a song of Spiro Agnew and all t...   
4     [If I look like home to you, if I am your sign...   
...                                                 ...   
7699  [Ancient holy wars, dead religions, holocausts...   
7700  [It's less hard than it should be to find a de...   
7701  [I found you once, and I'll find you again, Yo...   
7702  [Flyin' on past in your voodoo mask, High on y...   
7703  [We met in a parking lot, I was buying coffee ...   

                       song           artist  
0     theloveofcommonpeople       johndenver  
1     catchanotherbutterfly       johndenver  
2                  daydream       johndenver  
3     theballadofspiroagnew       johndenver  
4                    circus       johndenver  
...                     ...              ...  
7699      thisatombombandme  fatherjohnmisty  
7700          thisisamerica  fatherjohnmisty  
7701                    tor  fatherjohnmisty  
7702                    tos  fatherjohnmisty  
7703        wemetatthestore  fatherjohnmisty  

[7704 rows x 3 columns]

One possible thing I should check for is cover songs. I'll look for this by looking up song names that contain the word cover, as AZ-lyrics combines "cover" into the song name.

In [13]:
folk_df[folk_df['song'].str.contains('cover')]

lyrics  \
2731  [Well I'm going under and this time I fear the...   
2883  [Truth is you've been waiting so long, You're ...   
3873  [I had what I thought were clear open eyes, br...   
6730  [Hey baby, This is not a challenge, It just me...   

                                  song       artist  
2731  someoneyoulovedlewiscapaldicover    passenger  
2883                 undercoverofnight  arloguthrie  
3873              theundiscoveredfirst        feist  
6730                        discoverer          rem

It seems like the only real cover here is someone you loved cover by passenger. So i'll remove that.

In [5]:
folk_df = folk_df[~folk_df['song'].str.contains('lewiscapaldicover')]
folk_df

lyrics  \
0     [Living on free food tickets. Water in the mil...   
1     [Do you remember days not so very long ago, wh...   
2     [Yes, I been dreaming about you every day, eac...   
3     [I'll sing you a song of Spiro Agnew and all t...   
4     [If I look like home to you, if I am your sign...   
...                                                 ...   
7699  [Ancient holy wars, dead religions, holocausts...   
7700  [It's less hard than it should be to find a de...   
7701  [I found you once, and I'll find you again, Yo...   
7702  [Flyin' on past in your voodoo mask, High on y...   
7703  [We met in a parking lot, I was buying coffee ...   

                       song           artist  
0     theloveofcommonpeople       johndenver  
1     catchanotherbutterfly       johndenver  
2                  daydream       johndenver  
3     theballadofspiroagnew       johndenver  
4                    circus       johndenver  
...                     ...              ...  
7699      thisatombombandme  fatherjohnmisty  
7700          thisisamerica  fatherjohnmisty  
7701                    tor  fatherjohnmisty  
7702                    tos  fatherjohnmisty  
7703        wemetatthestore  fatherjohnmisty  

[7703 rows x 3 columns]

Now I'll check for remixes with the same str.contains method.

In [6]:
folk_df[folk_df['song'].str.contains('remix')]


lyrics  \
1515  [Nobody drives to the middle of nowhere, Just ...   
5846  [I'm drawn to the blood, The flight of a one-w...   
5847  [Spirit of my silence I can hear you, But I'm ...   
5849  [I'm drawn to the blood, The flight of a one-w...   
5851  [I'm Aporia's friend, Loneliness rides in my b...   
5852  [Shall we beat this or celebrate it?, You're n...   
5853  [The evil it spread like a fever ahead, It was...   
5883  [You're like a messiah kid, Little kingdoms in...   
6168  [[St. Vincent:], New York isn't New York witho...   
6171  [Sometimes I go to the edge of my roof, And I ...   
6187  [Happiness hit her [repeat], , Leave all your ...   
6228  [Now the sun is up and I'm going blind, Anothe...   
7361  [Downtown we let it go, Sunset high and our bo...   

                                      song              artist  
1515                          halfwayremix          jamesblunt  
5846     drawntothebloodsufjanstevensremix       sufjanstevens  
5847      deathwithdignityheladonegroremix       sufjanstevens  
5849     drawntothebloodfingerpickingremix       sufjanstevens  
5851            explodingwhaledovemanremix       sufjanstevens  
5852  allofmewantsallofyouheladonegroremix       sufjanstevens  
5853                 fourthofjuly900xremix       sufjanstevens  
5883           almostcrimesradiokillsremix   brokensocialscene  
6168                          newyorkremix           stvincent  
6171               smokingsectionjlinremix           stvincent  
6187           dogdaysareoveryeasayerremix  florencethemachine  
6228              delilahgalantisremixedit  florencethemachine  
7361                  loseitjerryfolkremix            ohwonder

These all seem to be remixes, so I'll go ahead and remove them now by assigning the dataframe to everything except these songs (using ~). 

In [7]:
folk_df = folk_df[~folk_df['song'].str.contains('remix')]
folk_df

lyrics  \
0     [Living on free food tickets. Water in the mil...   
1     [Do you remember days not so very long ago, wh...   
2     [Yes, I been dreaming about you every day, eac...   
3     [I'll sing you a song of Spiro Agnew and all t...   
4     [If I look like home to you, if I am your sign...   
...                                                 ...   
7699  [Ancient holy wars, dead religions, holocausts...   
7700  [It's less hard than it should be to find a de...   
7701  [I found you once, and I'll find you again, Yo...   
7702  [Flyin' on past in your voodoo mask, High on y...   
7703  [We met in a parking lot, I was buying coffee ...   

                       song           artist  
0     theloveofcommonpeople       johndenver  
1     catchanotherbutterfly       johndenver  
2                  daydream       johndenver  
3     theballadofspiroagnew       johndenver  
4                    circus       johndenver  
...                     ...              ...  
7699      thisatombombandme  fatherjohnmisty  
7700          thisisamerica  fatherjohnmisty  
7701                    tor  fatherjohnmisty  
7702                    tos  fatherjohnmisty  
7703        wemetatthestore  fatherjohnmisty  

[7690 rows x 3 columns]

Another cleaning step is dropping duplicates. To do this I can simply use pandas' drop_duplicates.

In [8]:
folk_df.drop_duplicates(subset="song", keep=False, inplace=True)

<ipython-input-8-0274d043e55d>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  folk_df.drop_duplicates(subset="song", keep=False, inplace=True)


In [9]:
folk_df

lyrics  \
0     [Living on free food tickets. Water in the mil...   
1     [Do you remember days not so very long ago, wh...   
2     [Yes, I been dreaming about you every day, eac...   
3     [I'll sing you a song of Spiro Agnew and all t...   
4     [If I look like home to you, if I am your sign...   
...                                                 ...   
7699  [Ancient holy wars, dead religions, holocausts...   
7700  [It's less hard than it should be to find a de...   
7701  [I found you once, and I'll find you again, Yo...   
7702  [Flyin' on past in your voodoo mask, High on y...   
7703  [We met in a parking lot, I was buying coffee ...   

                       song           artist  
0     theloveofcommonpeople       johndenver  
1     catchanotherbutterfly       johndenver  
2                  daydream       johndenver  
3     theballadofspiroagnew       johndenver  
4                    circus       johndenver  
...                     ...              ...  
7699      thisatombombandme  fatherjohnmisty  
7700          thisisamerica  fatherjohnmisty  
7701                    tor  fatherjohnmisty  
7702                    tos  fatherjohnmisty  
7703        wemetatthestore  fatherjohnmisty  

[6951 rows x 3 columns]

In order to look within the lyrics, I have to add another column which is the lyrics column converted to a string. That way I can actually look up the content within the lyrics.

In [10]:
folk_df['lyrics_string'] = folk_df['lyrics'].apply(lambda x: ','.join(map(str, x)))

<ipython-input-10-bdaa71140149>:1: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame.
Try using .loc[row_indexer,col_indexer] = value instead

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  folk_df['lyrics_string'] = folk_df['lyrics'].apply(lambda x: ','.join(map(str, x)))


In [11]:
folk_df

lyrics  \
0     [Living on free food tickets. Water in the mil...   
1     [Do you remember days not so very long ago, wh...   
2     [Yes, I been dreaming about you every day, eac...   
3     [I'll sing you a song of Spiro Agnew and all t...   
4     [If I look like home to you, if I am your sign...   
...                                                 ...   
7699  [Ancient holy wars, dead religions, holocausts...   
7700  [It's less hard than it should be to find a de...   
7701  [I found you once, and I'll find you again, Yo...   
7702  [Flyin' on past in your voodoo mask, High on y...   
7703  [We met in a parking lot, I was buying coffee ...   

                       song           artist  \
0     theloveofcommonpeople       johndenver   
1     catchanotherbutterfly       johndenver   
2                  daydream       johndenver   
3     theballadofspiroagnew       johndenver   
4                    circus       johndenver   
...                     ...              ...   
7699      thisatombombandme  fatherjohnmisty   
7700          thisisamerica  fatherjohnmisty   
7701                    tor  fatherjohnmisty   
7702                    tos  fatherjohnmisty   
7703        wemetatthestore  fatherjohnmisty   

                                          lyrics_string  
0     Living on free food tickets. Water in the milk...  
1     Do you remember days not so very long ago, whe...  
2     Yes, I been dreaming about you every day, each...  
3     I'll sing you a song of Spiro Agnew and all th...  
4     If I look like home to you, if I am your sign,...  
...                                                 ...  
7699  Ancient holy wars, dead religions, holocausts,...  
7700  It's less hard than it should be to find a dea...  
7701  I found you once, and I'll find you again,Your...  
7702  Flyin' on past in your voodoo mask,High on you...  
7703  We met in a parking lot,I was buying coffee an...  

[6951 rows x 4 columns]

I noticed that songs that contain 'Lyrics' within the lyrics themselves have serious data quality issues, with a lot of additional text and the beginning and end of the songs. So, I'll drop these, as I can't automatically correct them. 

In [12]:
folk_df[folk_df['lyrics_string'].str.contains('Lyrics')]

lyrics  \
41    [John Denver Lyrics, , "Take Me Home, Country ...   
336   [Kris Kristofferson Lyrics, , "Give It Time To...   
339   [Kris Kristofferson Lyrics, , "Why Me", Why me...   
558   ["Maybe I Think Too Much (A)", , They say that...   
770   [Yusuf Lyrics, , "Whispers From A Spiritual Ga...   
...                                                 ...   
7624  [Of Montreal Lyrics, , "Japanese Word For Witc...   
7628  [Of Montreal Lyrics, , "I Feel Safe With You, ...   
7630  [Of Montreal Lyrics, , "Notes Of Violate Spect...   
7633  [Of Montreal Lyrics, , "Extract The Masculine ...   
7646  [Father John Misty Lyrics, , "O I Long To Feel...   

                                         song             artist  \
41                     takemehomecountryroads         johndenver   
336                      giveittimetobetender  kriskristofferson   
339                                     whyme  kriskristofferson   
558                       maybeithinktoomucha          paulsimon   
770              whispersfromaspiritualgarden         catstevens   
...                                       ...                ...   
7624                     japanesewordforwitch         ofmontreal   
7628                    ifeelsafewithyoutrash         ofmontreal   
7630    notesofviolatespectatesaflatterofmale         ofmontreal   
7633  extractthemasculinegermfromremotememory         ofmontreal   
7646             oilongtofeelyourarmsaroundme    fatherjohnmisty   

                                          lyrics_string  
41    John Denver Lyrics,,"Take Me Home, Country Roa...  
336   Kris Kristofferson Lyrics,,"Give It Time To Be...  
339   Kris Kristofferson Lyrics,,"Why Me",Why me Lor...  
558   "Maybe I Think Too Much (A)",,They say that th...  
770   Yusuf Lyrics,,"Whispers From A Spiritual Garde...  
...                                                 ...  
7624  Of Montreal Lyrics,,"Japanese Word For Witch",...  
7628  Of Montreal Lyrics,,"I Feel Safe With You, Tra...  
7630  Of Montreal Lyrics,,"Notes Of Violate Spectate...  
7633  Of Montreal Lyrics,,"Extract The Masculine Ger...  
7646  Father John Misty Lyrics,,"O I Long To Feel Yo...  

[225 rows x 4 columns]

In [13]:
folk_df = folk_df[~folk_df['lyrics_string'].str.contains('Lyrics')]

In [14]:
folk_df

lyrics  \
0     [Living on free food tickets. Water in the mil...   
1     [Do you remember days not so very long ago, wh...   
2     [Yes, I been dreaming about you every day, eac...   
3     [I'll sing you a song of Spiro Agnew and all t...   
4     [If I look like home to you, if I am your sign...   
...                                                 ...   
7699  [Ancient holy wars, dead religions, holocausts...   
7700  [It's less hard than it should be to find a de...   
7701  [I found you once, and I'll find you again, Yo...   
7702  [Flyin' on past in your voodoo mask, High on y...   
7703  [We met in a parking lot, I was buying coffee ...   

                       song           artist  \
0     theloveofcommonpeople       johndenver   
1     catchanotherbutterfly       johndenver   
2                  daydream       johndenver   
3     theballadofspiroagnew       johndenver   
4                    circus       johndenver   
...                     ...              ...   
7699      thisatombombandme  fatherjohnmisty   
7700          thisisamerica  fatherjohnmisty   
7701                    tor  fatherjohnmisty   
7702                    tos  fatherjohnmisty   
7703        wemetatthestore  fatherjohnmisty   

                                          lyrics_string  
0     Living on free food tickets. Water in the milk...  
1     Do you remember days not so very long ago, whe...  
2     Yes, I been dreaming about you every day, each...  
3     I'll sing you a song of Spiro Agnew and all th...  
4     If I look like home to you, if I am your sign,...  
...                                                 ...  
7699  Ancient holy wars, dead religions, holocausts,...  
7700  It's less hard than it should be to find a dea...  
7701  I found you once, and I'll find you again,Your...  
7702  Flyin' on past in your voodoo mask,High on you...  
7703  We met in a parking lot,I was buying coffee an...  

[6726 rows x 4 columns]

Now we have 6726 songs left. I'll now look up songs that have square brackets in the lyrics. This is usually added to indicate song section or sometimes who is singing.

In [15]:
folk_df[folk_df['lyrics_string'].str.contains(']')]

lyrics  \
64    [[Summer], , Silently the morning mist is lyin...   
299   [[Spoken:], A couple of years back, I come acr...   
308   [Duvalier was a bitter man who cursed the morn...   
320   [I've been chasing after Josie since the day I...   
321   [I stand on the stairway, My back to the dunge...   
...                                                 ...   
7587  [Yeah, no blood hunger parasites, I'm inside o...   
7591  [Music stopped and I woke, , Somebody's paying...   
7636  [You should know, you should know that I, Am n...   
7638  [Ahhh..., It's you that I love, You're the onl...   
7674  [I'm gonna steal some bedsheets, From an amput...   

                                                   song             artist  \
64    seasonsuitesummerfallwinterlatewinterearlyspri...         johndenver   
299                                      tobeatthedevil  kriskristofferson   
308                                      duvaliersdream  kriskristofferson   
320                                               josie  kriskristofferson   
321                                     burdenoffreedom  kriskristofferson   
...                                                 ...                ...   
7587                                   monolithicegress         ofmontreal   
7591                         likeashokasinfernoofmemory         ofmontreal   
7636                                      firsttimehigh         ofmontreal   
7638                                         propaganda         ofmontreal   
7674                                            thememo    fatherjohnmisty   

                                          lyrics_string  
64    [Summer],,Silently the morning mist is lying o...  
299   [Spoken:],A couple of years back, I come acros...  
308   Duvalier was a bitter man who cursed the morni...  
320   I've been chasing after Josie since the day I ...  
321   I stand on the stairway,My back to the dungeon...  
...                                                 ...  
7587  Yeah, no blood hunger parasites,I'm inside of ...  
7591  Music stopped and I woke,,Somebody's paying at...  
7636  You should know, you should know that I,Am not...  
7638  Ahhh...,It's you that I love,You're the only t...  
7674  I'm gonna steal some bedsheets,From an amputee...  

[605 rows x 4 columns]

605 songs have this issue! Rather than removing them I can take advantage of the fact that I have the lyrics separated into lists of lines. Using the following for loop I can remove any array (comma separated value) which contains a square bracket. 

In [16]:
for lists in folk_df['lyrics']:
    for x in lists:
        if ']' in x:
            lists.remove(x)

Let's check one of the examples. Number 299 had [Spoken] at the beginning. Let's see if it fixed it.

In [17]:
folk_df['lyrics'][299]

["A couple of years back, I come across a great and wasted friend of mine in the hallway of a recording studio; and while he was reciting some poetry to me that he'd written, I saw that he was about a step away from dyin' and I couldn't help but wonder why. And the lines of this song occurred to me. I'm happy to say he's no longer wasted and he's got him a good woman. And I'd like to dedicate this to John and June, who helped show me how to beat the devil.",
 '',
 'It was winter time in Nashville, down on music city row.',
 "And I was lookin' for a place to get myself out of the cold.",
 "To warm the frozen feelin' that was eatin' at my soul.",
 'Keep the chilly wind off my guitar.',
 '',
 'My thirsty wanted whisky; my hungry needed beans,',
 "But it'd been of month of paydays since I'd heard that eagle scream.",
 'So with a stomach full of empty and a pocket full of dreams,',
 'I left my pride and stepped inside a bar.',
 '',
 "Actually, I guess you'd could call it a Tavern:",
 'Cigar

Looks good!

Now I am going to remove lines that contain potentially offensive language (i.e. common swear words, words associated with sexism and racism). I have replaced such language with X in the following code. 

In [18]:
folk_df[folk_df['lyrics_string'].str.contains('X')]

lyrics               song  \
584   [It was the morning of October 6th, 1960, 1 wa...      adioshermanos   
1435  [Diabolic pressure., It keeps getting worse be...   notworkingforyou   
1580  [Well, the moon is yellow and the people are t...          chinatown   
2931  [Lord, in a bourgeois town, It's a bourgeois t...  thebourgeoisblues   
6899  [Kinda cold and I'm with someone (na na na na ...              2late   
6906  [Benjamins baby, Benjamins (Yeah), Benjamins b...          benjiminz   

           artist                                      lyrics_string  
584     paulsimon  [SAL],It was the morning of October 6th, 1960,...  
1435  vanmorrison  Diabolic pressure.,It keeps getting worse befo...  
1580    johnprine  Well, the moon is yellow and the people are to...  
2931  arloguthrie  Lord, in a bourgeois town,It's a bourgeois tow...  
6899     toroymoi  Kinda cold and I'm with someone (na na na na n...  
6906     toroymoi  Benjamins baby,Benjamins (Yeah),Benjamins baby...

In [19]:
for lists in folk_df['lyrics']:
    for x in lists:
        if 'X' in x:
            lists.remove(x)

In [20]:
folk_df[folk_df['lyrics_string'].str.contains('X')]

lyrics                  song  \
5984  [In the sleepy west ofthe woody east, is a val...                 umass   
7183  [We're out looking for astronauts, Looking for...  lookingforastronauts   
7557  [There's not really a name for what we are, Or...   obviousatonicnuncio   

          artist                                      lyrics_string  
5984      pixies  In the sleepy west ofthe woody east,is a valle...  
7183    national  We're out looking for astronauts,Looking for a...  
7557  ofmontreal  There's not really a name for what we are,Or h...

In [21]:
for lists in folk_df['lyrics']:
    for x in lists:
        if 'X' in x:
            lists.remove(x)

In [22]:
folk_df[folk_df['lyrics_string'].str.contains('X')]

lyrics  \
434   [Here's a song about a good 'ole girl with a b...   
588   [Well, did you bring me my money,, My cab fare...   
896   [A Cosmopolitan a Manhattan, Call me one, Pour...   
1242  [Man makes his money and they call him rich, D...   
1651  [Feeling kind of bony, On the telephoney, Talk...   
1941  [Betty Jo, you made it plain, when first you h...   
2600  [The day you left and you called me bitch, I c...   
2866  [Gonna sit right here and sing myself a tune, ...   
3077  [He was sitting in the lounge of the Empire Ho...   
3081  [Rousseau walks on trumpet paths, Safaris to t...   
3995  [Electricity!, , Took some lightning, a kite, ...   
5085  [You're a rich girl and you're goin' too far, ...   
5444  [It's coming through a hole in the air, From t...   
5550  [I see the lightning almost every day, I've st...   
5757  [Flash that buttery gold, jittery zeitgeist, W...   
5877  [Tonya Harding, my star, Well this world is a ...   
5907  [Texico bitches, have you begun again?, Texico...   
5911  [Ungrateful little motherfuck, boredom tuck, B...   
5933  [How's it goin'?, I just wanted to check in, N...   
5970  [i know you're rich in, good clothes and littl...   
6048  [1-2-3-4, , My father lives in Arabia, And I l...   
6050  [Oh Hannah, I wanna feel you close, Oh Hannah,...   
6224  [I believe in you and in our hearts we know th...   
6905  [I don't want you anymore, , Do what you want,...   
7502  [Scientists are nihilists with good imaginatio...   
7516  [I'm a motherfuckin' headliner, bitch, you don...   
7552  [The sacrifice must be for some unknwon deity,...   
7558  [I've been trying to be your friend forever, I...   
7559  [I'm the wrong color but I'm on the one, Wish ...   
7590  [Crowd and knitter of killers, With their wild...   
7593  [It's different for girls, They've got sensiti...   
7614  [Went to her art show, "Hello", "Hello", "Yeah...   

                               song              artist  \
434                  downtohersocks   kriskristofferson   
588                     thevampires           paulsimon   
896                       ididitall        tracychapman   
1242               manhastostruggle         vanmorrison   
1651               leavethelightson           johnprine   
1941                bettycalledmein     gordonlightfoot   
2600                  medicinewheel           aimeemann   
2866                            eli         arloguthrie   
3077                raisedonrobbery        jonimitchell   
3081                  thejungleline        jonimitchell   
3995               benfranklinssong        decemberists   
5085                       richgirl          ninasimone   
5444                      democracy         judycollins   
5550                      theyearis          arcadefire   
5757      starofwondernoneshallpass       sufjanstevens   
5877                   tonyaharding       sufjanstevens   
5907                  texicobitches   brokensocialscene   
5911         ungratefullittlefather   brokensocialscene   
5933                letstrytheafter   brokensocialscene   
5970                     issheweird              pixies   
6048                weirdatmyschool              pixies   
6050         iwannabeyourgirlfriend           girlinred   
6224                       100years  florencethemachine   
6905                           want            toroymoi   
7502                 gronlandicedit          ofmontreal   
7516                   wickedwisdom          ofmontreal   
7552            yerenewtheplaintiff          ofmontreal   
7558            sailshermaphroditic          ofmontreal   
7559                     steppinout          ofmontreal   
7590  chthoniandirgeforuruktheother          ofmontreal   
7593           itsdifferentforgirls          ofmontreal   
7614                   polyaneurism          ofmontreal   

                                          lyrics_string  
434   Here's a song about a good 'ole girl with a bi...  
588   [HERNANDEZ],Well, did you bring

In [23]:
for lists in folk_df['lyrics']:
    for x in lists:
        if 'X' in x:
            lists.remove(x)

In [24]:
folk_df[folk_df['lyrics_string'].str.contains('X')]

lyrics               song  \
7528  [(Ahh, yeah!), (Is that my Concorde on the lau...          idengager   
7559  [I'm the wrong color but I'm on the one, Wish ...         steppinout   
7662  [Ancient holy wars, Dead religions, holocausts...           holyshit   
7699  [Ancient holy wars, dead religions, holocausts...  thisatombombandme   

               artist                                      lyrics_string  
7528       ofmontreal  (Ahh, yeah!),(Is that my Concorde on the launc...  
7559       ofmontreal  I'm the wrong color but I'm on the one,Wish I ...  
7662  fatherjohnmisty  Ancient holy wars,Dead religions, holocausts,N...  
7699  fatherjohnmisty  Ancient holy wars, dead religions, holocausts,...

In [25]:
folk_df.drop(7699, inplace=True)

/Users/liamgentile/opt/anaconda3/envs/deeplearning/lib/python3.8/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


In [26]:
for lists in folk_df['lyrics']:
    for x in lists:
        if 'X' in x:
            lists.remove(x)

In [27]:
folk_df[folk_df['lyrics_string'].str.contains('X')]

lyrics  \
889   [Say I have known some, Less than I should, Sa...   
1449  [Judging by the look on the organ-grinder, He'...   
1514  [And don't forget that I'll be watching you, A...   
3608  [I'll shoot you dead, For the father, Of the c...   
3925  [We are two mariners, Our ships' sole survivor...   
4471  [I don’t know you to wear a hat, But I came ho...   
5642  [A dreaded sunny day, So I meet you at the cem...   
5671  [At the record company meeting, On their hands...   
5674  [We are a servant, we have a song, The side of...   
5757  [Flash that buttery gold, jittery zeitgeist, W...   
5901  [It's all gonna break., when i was a kid, you ...   
5903  [All my friends in magazines, Got addicted to ...   
5906  [All the dirty fingers picking up finds, Bruis...   
5943  [One two three, She's a real left winger 'caus...   
5965  [Hey, Been trying to meet you, Hey, Must be a ...   
6299  [I feel a fading here, I feel it everyday, It'...   
6786  [I could have it, Without with the whores and ...   
7559  [I'm the wrong color but I'm on the one, Wish ...   

                           song             artist  \
889                  neveryours       tracychapman   
1449                outofmymind         jamesblunt   
1514                     dontgo         jamesblunt   
3608                 icicletusk         fleetfoxes   
3925     themarinersrevengesong       decemberists   
4471                   civilwar           joanbaez   
5642               cemetrygates             smiths   
5671        paintavulgarpicture             smiths   
5674            wearewhatyousay      sufjanstevens   
5757  starofwondernoneshallpass      sufjanstevens   
5901           itsallgonnabreak  brokensocialscene   
5903               allmyfriends  brokensocialscene   
5906                 chasescene  brokensocialscene   
5943               ivebeentired             pixies   
5965                        hey             pixies   
6299                       acid   neutralmilkhotel   
6786               brokenbreads   newpornographers   
7559                 steppinout         ofmontreal   

                                          lyrics_string  
889   Say I have known some,Less than I should,Say I...  
1449  Judging by the look on the organ-grinder,He'll...  
1514  And don't forget that I'll be watching you,And...  
3608  I'll shoot you dead,For the father,Of the coal...  
3925  We are two mariners,Our ships' sole survivors,...  
4471  I don’t know you to wear a hat,But I came home...  
5642  A dreaded sunny day,So I meet you at the cemet...  
5671  At the record company meeting,On their hands -...  
5674  We are a servant, we have a song,The side of a...  
5757  Flash that buttery gold, jittery zeitgeist,Wit...  
5901  It's all gonna break.,when i was a kid, you fu...  
5903  All my friends in magazines,Got addicted to th...  
5906  All the dirty fingers picking up finds,Bruised...  
5943  One two three,She's a real left winger 'cause ...  
5965  Hey,Been trying to meet you,Hey,Must be a devi...  
6299  I feel a fading here,I feel it everyday,It's j...  
6786  I could have it,Without with the whores and th...  
7559  I'm the wrong color but I'm on the one,Wish I ...

In [28]:
for lists in folk_df['lyrics']:
    for x in lists:
        if 'X' in x:
            lists.remove(x)

Now that I've removed duplicates, covers, remixes, and songs with serious quality issues, as well as individual lines with non-lyric material and potentially offensive language, I can write the cleaned dataframe to a new csv.

In [29]:
folk_df.to_csv('/Users/liamgentile/Desktop/folk_df.csv')

------

# Pop lyrics

In [30]:
pop_df = pd.read_csv('/Users/liamgentile/Desktop/lyric_files/pop_lyrics.csv', converters={'lyrics': eval})

In [31]:
pop_df.head(5)

Unnamed: 0                                             lyrics  \
0           0  [He said the way my blue eyes shined, Put thos...   
1           1  [State the obvious, I didn't get my perfect fa...   
2           2  [Taylor Swift Lyrics, , "Teardrops On My Guita...   
3           3  [Taylor Swift Lyrics, , "A Place In This World...   
4           4  [You have a way of coming easily to me, And wh...   

                  song       artist  
0            timmcgraw  taylorswift  
1        picturetoburn  taylorswift  
2  teardropsonmyguitar  taylorswift  
3    aplaceinthisworld  taylorswift  
4            coldasyou  taylorswift

Again, with the pop lyric dataframe we have an additional, unnecessary index column which I will drop.

In [32]:
pop_df.drop('Unnamed: 0', axis=1, inplace=True)
pop_df

lyrics  \
0     [He said the way my blue eyes shined, Put thos...   
1     [State the obvious, I didn't get my perfect fa...   
2     [Taylor Swift Lyrics, , "Teardrops On My Guita...   
3     [Taylor Swift Lyrics, , "A Place In This World...   
4     [You have a way of coming easily to me, And wh...   
...                                                 ...   
6963  [Sorry I'm not so merry, But I feel like this ...   
6964  [Straight up, Tell me everything you've been t...   
6965  [[Yungen:], Young Mabes, Yungen, One shot, , L...   
6966  [I'm thinking of you, I'm thinking of you, I'm...   
6967  [[Mabel:], Why can't I tell if you're into me,...   

                           song       artist  
0                     timmcgraw  taylorswift  
1                 picturetoburn  taylorswift  
2           teardropsonmyguitar  taylorswift  
3             aplaceinthisworld  taylorswift  
4                     coldasyou  taylorswift  
...                         ...          ...  
6963        loneliesttimeofyear        mabel  
6964                myboymytown        mabel  
6965               oneshotremix        mabel  
6966              thinkingofyou        mabel  
6967  thinkingofyoucadenzaremix        mabel  

[6968 rows x 3 columns]

I'll start by creating the lyrics string column in order to be able to look up specific content within lyrics (as the lyrics column is structured in the form of lists.

In [33]:
pop_df['lyrics_string'] = pop_df['lyrics'].apply(lambda x: ','.join(map(str, x)))
pop_df

lyrics  \
0     [He said the way my blue eyes shined, Put thos...   
1     [State the obvious, I didn't get my perfect fa...   
2     [Taylor Swift Lyrics, , "Teardrops On My Guita...   
3     [Taylor Swift Lyrics, , "A Place In This World...   
4     [You have a way of coming easily to me, And wh...   
...                                                 ...   
6963  [Sorry I'm not so merry, But I feel like this ...   
6964  [Straight up, Tell me everything you've been t...   
6965  [[Yungen:], Young Mabes, Yungen, One shot, , L...   
6966  [I'm thinking of you, I'm thinking of you, I'm...   
6967  [[Mabel:], Why can't I tell if you're into me,...   

                           song       artist  \
0                     timmcgraw  taylorswift   
1                 picturetoburn  taylorswift   
2           teardropsonmyguitar  taylorswift   
3             aplaceinthisworld  taylorswift   
4                     coldasyou  taylorswift   
...                         ...          ...   
6963        loneliesttimeofyear        mabel   
6964                myboymytown        mabel   
6965               oneshotremix        mabel   
6966              thinkingofyou        mabel   
6967  thinkingofyoucadenzaremix        mabel   

                                          lyrics_string  
0     He said the way my blue eyes shined,Put those ...  
1     State the obvious, I didn't get my perfect fan...  
2     Taylor Swift Lyrics,,"Teardrops On My Guitar",...  
3     Taylor Swift Lyrics,,"A Place In This World",,...  
4     You have a way of coming easily to me,And when...  
...                                                 ...  
6963  Sorry I'm not so merry,But I feel like this ye...  
6964  Straight up,Tell me everything you've been thr...  
6965  [Yungen:],Young Mabes,Yungen,One shot,,Look,If...  
6966  I'm thinking of you,I'm thinking of you,I'm th...  
6967  [Mabel:],Why can't I tell if you're into me,I'...  

[6968 rows x 4 columns]

Let's see how many remixes there are...

In [34]:
pop_df[pop_df['song'].str.contains('remix')]

lyrics  \
164   [[Taylor Swift:], ‘Cause, baby, now we've got ...   
189   [Once upon a time a few mistakes ago, I was in...   
200   [We were both young when I first saw you, I cl...   
203   [Taylor Swift Lyrics, , "Lover (First Dance Re...   
204   [We could leave the Christmas lights up 'til J...   
...                                                 ...   
6923  [And now, breathe deep, I'm inhaling, You and ...   
6945  [They call me Burna...Boy, E-E-E-Eee, Yeah, Ye...   
6960  [[Mabel:], We came in crawling this morning, A...   
6965  [[Yungen:], Young Mabes, Yungen, One shot, , L...   
6967  [[Mabel:], Why can't I tell if you're into me,...   

                                    song        artist  \
164                        badbloodremix   taylorswift   
189             iknewyouweretroubleremix   taylorswift   
200   lovestorytaylorsversionelviraremix   taylorswift   
203                 loverfirstdanceremix   taylorswift   
204                           loverremix   taylorswift   
...                                  ...           ...   
6923                     alaskasohnremix  maggierogers   
6945                 finderskeepersremix         mabel   
6960                       finelineremix         mabel   
6965                        oneshotremix         mabel   
6967           thinkingofyoucadenzaremix         mabel   

                                          lyrics_string  
164   [Taylor Swift:],‘Cause, baby, now we've got ba...  
189   Once upon a time a few mistakes ago,I was in y...  
200   We were both young when I first saw you,I clos...  
203   Taylor Swift Lyrics,,"Lover (First Dance Remix...  
204   We could leave the Christmas lights up 'til Ja...  
...                                                 ...  
6923  And now, breathe deep,I'm inhaling,You and I, ...  
6945  They call me Burna...Boy,E-E-E-Eee,Yeah,Yeah-y...  
6960  [Mabel:],We came in crawling this morning,Alre...  
6965  [Yungen:],Young Mabes,Yungen,One shot,,Look,If...  
6967  [Mabel:],Why can't I tell if you're into me,I'...  

[214 rows x 4 columns]

214! These should be removed as they are likely duplicates.

In [35]:
pop_df = pop_df[~pop_df['song'].str.contains('remix')]

Now let's look for covers.

In [36]:
pop_df[pop_df['song'].str.contains('cover')]

lyrics               song  \
643   [First I'll acknowledge, Your trust has been b...           recovery   
1207  [I look at you, and all I wanna do, is just di...         undercover   
2595  [Pink & Willow Sage Hart Lyrics, , "Cover Me I...  covermeinsunshine   
3698  [There's lights that line these halls, That be...            coverup   
6004  [No one, no one, no one,, Can get in the way o...    7acousticcovers   
6734  [James Arthur Lyrics, , "Recovery", , I don't ...           recovery   

               artist                                      lyrics_string  
643      justinbieber  First I'll acknowledge,Your trust has been bro...  
1207      selenagomez  I look at you, and all I wanna do, is just dis...  
2595             pink  Pink & Willow Sage Hart Lyrics,,"Cover Me In S...  
3698   imaginedragons  There's lights that line these halls,That beck...  
6004  twentyonepilots  No one, no one, no one,,Can get in the way of ...  
6734      jamesarthur  James Arthur Lyrics,,"Recovery",,I don't want ...

Not too bad, looks like the only real cover is 7acousticcovers, which I'll go ahead and remove.

In [37]:
pop_df = pop_df[~pop_df['song'].str.contains('7acousticcovers')]

Now, just like with the folk_df, I'll remove songs that have serious data quality issues (these have 'lyrics' in them) and have a lot of extra non-lyrical content.

In [38]:
pop_df[pop_df['lyrics_string'].str.contains('Lyrics')]

lyrics  \
2     [Taylor Swift Lyrics, , "Teardrops On My Guita...   
3     [Taylor Swift Lyrics, , "A Place In This World...   
5     [Taylor Swift Lyrics, , "The Outside", , I did...   
6     [Taylor Swift Lyrics, , "Tied Together With A ...   
8     [Taylor Swift Lyrics, , "Should've Said No", ,...   
...                                                 ...   
6839  [Banks Lyrics, , "Stroke", , You want me, You ...   
6842  [Banks Lyrics, , "Look What You're Doing To Me...   
6867  [Jessie Reyez Lyrics, , "Imported", (feat. JRM...   
6884  [Jessie Reyez Lyrics, , "Crazy", (originally b...   
6924  [Maggie Rogers Lyrics, , "Love You For A Long ...   

                        song        artist  \
2        teardropsonmyguitar   taylorswift   
3          aplaceinthisworld   taylorswift   
5                 theoutside   taylorswift   
6     tiedtogetherwithasmile   taylorswift   
8             shouldvesaidno   taylorswift   
...                      ...           ...   
6839                  stroke         banks   
6842  lookwhatyouredoingtome         banks   
6867                imported   jessiereyez   
6884                   crazy   jessiereyez   
6924     loveyouforalongtime  maggierogers   

                                          lyrics_string  
2     Taylor Swift Lyrics,,"Teardrops On My Guitar",...  
3     Taylor Swift Lyrics,,"A Place In This World",,...  
5     Taylor Swift Lyrics,,"The Outside",,I didn't k...  
6     Taylor Swift Lyrics,,"Tied Together With A Smi...  
8     Taylor Swift Lyrics,,"Should've Said No",,It's...  
...                                                 ...  
6839  Banks Lyrics,,"Stroke",,You want me,You want m...  
6842  Banks Lyrics,,"Look What You're Doing To Me",(...  
6867  Jessie Reyez Lyrics,,"Imported",(feat. JRM),,[...  
6884  Jessie Reyez Lyrics,,"Crazy",(originally by Wi...  
6924  Maggie Rogers Lyrics,,"Love You For A Long Tim...  

[986 rows x 4 columns]

In [39]:
pop_df = pop_df[~pop_df['lyrics_string'].str.contains('Lyrics')]
pop_df

lyrics  \
0     [He said the way my blue eyes shined, Put thos...   
1     [State the obvious, I didn't get my perfect fa...   
4     [You have a way of coming easily to me, And wh...   
7     [Cory's eyes are like a jungle, He smiles, it'...   
11    [Last Christmas I gave you my heart, But the v...   
...                                                 ...   
6961  [(I'll be home, I'll be home), , I'm dreaming ...   
6962  [Lose a layer or two and let's get lost in tim...   
6963  [Sorry I'm not so merry, But I feel like this ...   
6964  [Straight up, Tell me everything you've been t...   
6966  [I'm thinking of you, I'm thinking of you, I'm...   

                       song       artist  \
0                 timmcgraw  taylorswift   
1             picturetoburn  taylorswift   
4                 coldasyou  taylorswift   
7             staybeautiful  taylorswift   
11            lastchristmas  taylorswift   
...                     ...          ...   
6961  illbehomeforchristmas        mabel   
6962           knowmebetter        mabel   
6963    loneliesttimeofyear        mabel   
6964            myboymytown        mabel   
6966          thinkingofyou        mabel   

                                          lyrics_string  
0     He said the way my blue eyes shined,Put those ...  
1     State the obvious, I didn't get my perfect fan...  
4     You have a way of coming easily to me,And when...  
7     Cory's eyes are like a jungle,He smiles, it's ...  
11    Last Christmas I gave you my heart,But the ver...  
...                                                 ...  
6961  (I'll be home, I'll be home),,I'm dreaming ton...  
6962  Lose a layer or two and let's get lost in time...  
6963  Sorry I'm not so merry,But I feel like this ye...  
6964  Straight up,Tell me everything you've been thr...  
6966  I'm thinking of you,I'm thinking of you,I'm th...  

[5767 rows x 4 columns]

We now have 5767 rows left. Now let's see how many songs have a line with square brackets indicating non-lyrical content.

In [40]:
pop_df[pop_df['lyrics_string'].str.contains(']')]

lyrics  \
1     [State the obvious, I didn't get my perfect fa...   
7     [Cory's eyes are like a jungle, He smiles, it'...   
22    [You're on the phone with your girlfriend, She...   
31    [I am not the kind of girl, Who should be rude...   
43    [[Verse 1:], Loving him is like driving a new ...   
...                                                 ...   
6886  [[Jessie Reyez:], Figures, I'm the bad guy cau...   
6888  [Dub, , [Jessie Reyez:], Ponme el nombre que t...   
6932  [[Kojo Funds:], Yeah, Yo yo yo yo, Mabel, uh, ...   
6937  [[Mabel:], We came in crawling this morning, A...   
6950  [[Mabel:], I don't want a selfish love no more...   

                          song       artist  \
1                picturetoburn  taylorswift   
7                staybeautiful  taylorswift   
22             youbelongwithme  taylorswift   
31                    speaknow  taylorswift   
43                         red  taylorswift   
...                        ...          ...   
6886           figuresareprise  jessiereyez   
6888  importedspanglishversion  jessiereyez   
6932            finderskeepers        mabel   
6937                  fineline        mabel   
6950               selfishlove        mabel   

                                          lyrics_string  
1     State the obvious, I didn't get my perfect fan...  
7     Cory's eyes are like a jungle,He smiles, it's ...  
22    You're on the phone with your girlfriend,She's...  
31    I am not the kind of girl,Who should be rudely...  
43    [Verse 1:],Loving him is like driving a new Ma...  
...                                                 ...  
6886  [Jessie Reyez:],Figures,I'm the bad guy cause ...  
6888  Dub,,[Jessie Reyez:],Ponme el nombre que tú qu...  
6932  [Kojo Funds:],Yeah,Yo yo yo yo,Mabel, uh,,Mama...  
6937  [Mabel:],We came in crawling this morning,Alre...  
6950  [Mabel:],I don't want a selfish love no more,W...  

[1447 rows x 4 columns]

1447! I'll go ahead and remove this square bracket wrapped content using a nested for loop to access specific items in the list that makes up a song in the lyrics column.

In [41]:
for array in pop_df['lyrics']:
    for x in array:
        if ']' in x:
            array.remove(x)

Let's look at an example. Song #43 has [Verse 1] at the beginning, let's see if it was removed.

In [42]:
pop_df['lyrics'][43]

['Loving him is like driving a new Maserati down a dead-end street',
 'Faster than the wind, passionate as sin, ending so suddenly',
 "Loving him is like trying to change your mind once you're already flying through the free fall",
 'Like the colors in autumn, so bright just before they lose it all',
 '',
 "Losing him was blue like I'd never known",
 'Missing him was dark grey all alone',
 'Forgetting him was like trying to know somebody you never met',
 'But loving him was red',
 'Loving him was red',
 '',
 'Touching him was like realizing all you ever wanted was right there in front of you',
 'Memorizing him was as easy as knowing all the words to your old favorite song',
 "Fighting with him was like trying to solve a crossword and realizing there's no right answer",
 'Regretting him was like wishing you never found out that love could be that strong',
 '',
 "Losing him was blue like I'd never known",
 'Missing him was dark grey all alone',
 'Forgetting him was like trying to know so

Looks good!

I noticed that there are some spanish language songs in the dataset. I'll go ahead now and search for some common spanish words. 

In [43]:
pop_df[pop_df['lyrics_string'].str.contains('él')]

lyrics  \
594   [(Alright let's go), , There's gonna be one le...   
1240  [Bebé, no sé si hablas mucho español, Si entie...   

                                   song        artist  \
594   onelesslonelygirlfrenchadaptation  justinbieber   
1240                       bailaconmigo   selenagomez   

                                          lyrics_string  
594   (Alright let's go),,There's gonna be one less ...  
1240  Bebé, no sé si hablas mucho español,Si entiend...

In [44]:
pop_df = pop_df[~pop_df['lyrics_string'].str.contains('él')]

I removed two songs that contained él. 

In [45]:
pop_df[pop_df['lyrics_string'].str.contains('soy')]

lyrics  \
1084  [American life, American dream, American life,...   
1108  [Yo te quiero means I love you, Mucho gusto me...   
1145  [Madonna, Maluma, , Bitch, I'm loca (Y yo loco...   
1242  [Nadie tiene que decírmelo, Hablas con otra qu...   
1627  [Hoy me acusas otra vez, Nada lograr que me, P...   
1728  [He used to call me DN, That stood for deadly ...   
4419  [If U wanna we can kick it, baby, joint 2 join...   
5057  [One day we were at home rehearsing, When we h...   
6864  [Te conviene no decirme que me quieres, Prefie...   

                                song         artist  \
1084  americanlifeheadcleanerrockmix        madonna   
1108                   spanishlesson        madonna   
1145                     bitchimloca        madonna   
1242                     selfishlove    selenagomez   
1627   tellmeyoulovemespanishversion     demilovato   
1728                   ultraviolence     lanadelrey   
4419                     joint2joint         prince   
5057                        year2019  jonasbrothers   
6864                            sola    jessiereyez   

                                          lyrics_string  
1084  American life,American dream,American life,Ame...  
1108  Yo te quiero means I love you,Mucho gusto mean...  
1145  Madonna, Maluma,,Bitch, I'm loca (Y yo loco, l...  
1242  Nadie tiene que decírmelo,Hablas con otra que ...  
1627  Hoy me acusas otra vez,Nada lograr que me,Perd...  
1728  He used to call me DN,That stood for deadly ni...  
4419  If U wanna we can kick it, baby, joint 2 joint...  
5057  One day we were at home rehearsing,When we hea...  
6864  Te conviene no decirme que me quieres,Prefiero...

In [46]:
pop_df.drop([1108, 1145, 1242, 1627, 6864], inplace=True)

/Users/liamgentile/opt/anaconda3/envs/deeplearning/lib/python3.8/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


I removed 5 songs that contained soy.

In [47]:
pop_df[pop_df['lyrics_string'].str.contains('tengo')]

lyrics                 song  \
1238  [Ya no duele como antes, no, La herida de tu a...             deunavez   
1241  [Tus labios son mi vicio, Por un beso tuyo tod...                vicio   
2149  [I feel it, Can you feel it tonight?, I feel t...         nothingonyou   
4930  [This is the rhythm, rhythm, rhythm, rhythm, T...  ritmobadboysforlife   
6882  [Peleamos otra vez, La casa está encendida, Si...          conelviento   

             artist                                      lyrics_string  
1238    selenagomez  Ya no duele como antes, no,La herida de tu amo...  
1241    selenagomez  Tus labios son mi vicio,Por un beso tuyo todo ...  
2149      edsheeran  I feel it,Can you feel it tonight?,I feel the ...  
4930  blackeyedpeas  This is the rhythm, rhythm, rhythm, rhythm,Thi...  
6882    jessiereyez  Peleamos otra vez,La casa está encendida,Si me...

In [48]:
pop_df.drop([1238, 1241, 6882], inplace=True)

/Users/liamgentile/opt/anaconda3/envs/deeplearning/lib/python3.8/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


I removed three songs that contain tengo.

In [49]:
pop_df[pop_df['lyrics_string'].str.contains('amor')]

lyrics  \
285   [It doesn't matter if you love him, or capital...   
290   [Love is just a history that they may prove, A...   
404   [Listen, I've got the sickest ambition, Deep i...   
427   [Oh oh oh, Welcome to the Panty Circus!, Oh oh...   
443   [Hold me daddy, real tight., They'll take our ...   
478   [I love the way you make me feel, I love it, I...   
622   [Hey, what's the situation, whoa?, I'm just tr...   
635   [Hey, what's the situation, whoa?, I'm just tr...   
740   [Travelled all across the world,, Seen a lot o...   
914   [Dime si me echas de menos aún, Dime si no me ...   
1007  [, , , , , , , , , Y mucho mas que eso, mi ser...   
1010  [Now Eva Peron had every disadvantage, You nee...   
1144  [Come baby, come, Come, vem cá, , Come, come t...   
1239  [Salen pa' que la vean, Se pierden en el ritmo...   
1747  [All the lights in Miami begin to gleam, Ruby,...   
1836  [What you don't tell no one you can tell me, L...   
1875  [Hollywood and New York, mister major, And the...   
1876  [One day my hoops will be made of diamonds, An...   
1893  [I was born bad, But then I met you, You made ...   
1916  [Walking is an art, so is my body, Papi is a w...   
1954  [Oh, oh, come on come on, I hope you remember ...   
2149  [I feel it, Can you feel it tonight?, I feel t...   
2180  [I found a love for me, Oh darling, just dive ...   
2604  [You lift my heart up when the rest of me is d...   
3047  [On my way to sunny California, On my way to s...   
3680  [Na, na, na, na, ooh, , Put it in drive, I'll ...   
4035  [Dov'e se adesso, Dov'e se, amore mio, , Dov'e...   
4054  [Every morning I would wake up, And I'd tie th...   
4467  [Mary, Claire, Denise and Belle, Best friends ...   
4658  [Don't you get me, , She wears a long fur coat...   
4769  [Que palabras te puedo decir, Que te hagan vol...   
4782  [Cuando sientas tristeza, Que no puedas calmar...   
4897  [I can't stand another night, Stand another ni...   
5246  [Are you ready?, , If you ain't got no money t...   
5254  [Life's a movie, let the camera roll, Fast lif...   
5266  [Life's a movie, let the camera roll, Fast lif...   
5587  [Shake that ass girl!, , Fogging up my glasses...   
5611  [Four, three, two, one, , Boys and hungry hear...   
5967  ["Ride", , I just wanna stay in the sun where ...   
6330  [My way in London, Americana, The persona, it'...   
6351  [Been a long time I ain't been in love, Little...   
6446  [Yeah, Yeah, hahaha, , I just wanna go real ha...   
6602  [Baby I, I got something to say t-t-tonight, W...   
6874  [Este amor me dejó sorda, Los consejos de mi m...   
6920  [You said, "I can't take no more", Lying by th...   

                                song           artist  \
285                      bornthisway         ladygaga   
290                       bloodymary         ladygaga   
404                        hollywood         ladygaga   
427                       pantyparty         ladygaga   
443                         sexyugly         ladygaga   
478           thewayspanglishversion     arianagrande   
622                          takeyou     justinbieber   
635           takeyouacousticversion     justinbieber   
740                           miamor     justinbieber   
914                    lovasaolvidar     billieeilish   
1007  acinemainbuenosaires26july1952          madonna   
1010     onthisnightofathousandstars          madonna   
1144                      fazgostoso          madonna   
1239                    buscandoamor      selenagomez   
1747                       salvatore       lanadelrey   
1836                   cherryblossom       lanadelrey   
1875                       hitandrun       lanadelrey   
1876                       hollywood       lanadelrey   
1893                  kindaouttaluck       lanadelrey   
1916                            noir       lanadelrey   
1954                televisionheaven       lanadelrey   
2149                    nothingonyou        edsheeran   
2180                 pe

In [50]:
pop_df.drop([478, 914, 1007, 1144, 1239, 4769, 4782, 6874], inplace=True)

/Users/liamgentile/opt/anaconda3/envs/deeplearning/lib/python3.8/site-packages/pandas/core/frame.py:4308: SettingWithCopyWarning: 
A value is trying to be set on a copy of a slice from a DataFrame

See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy
  return super().drop(


I removed 8 songs that contain amor.

Moving on, I noticed that some songs contain "version" in the title, I think this indicates duplicates as they are simple different versions of another song.

In [51]:
pop_df[pop_df['song'].str.contains('version')]

lyrics  \
135   [We were both young when I first saw you, I cl...   
143   [I'm five years old, it's getting cold, I've g...   
148   [This is wrong but I can't help but feel like,...   
149   [In the heat of the fight I walked away, Ignor...   
409   [Take my hand, stay Joanne, Heaven's not, Read...   
632   [As long as you love me, , We're under pressur...   
634   [She don't like the flash, wanna keep us in th...   
635   [Hey, what's the situation, whoa?, I'm just tr...   
637   [You're beautiful, beautiful, you should know ...   
1662  [Flippin' through all of these magazines, Tell...   
1670  [You wanna play, you wanna stay, you wanna hav...   
1924  [They all know my name now,, it's just another...   
1951  [Summertime, And the livin' is easy, Fish are ...   
1962  [All that's real to me is Marilyn and Jesus,, ...   
2722  [[Jonny:] So I said to her, "Angelina, I want ...   
2859  [, You are my only hope, The truth that dealt ...   
2860  [Could have given me something, You my everyth...   
3168  [I've been in this town so long that back in t...   
4080  [Chiquitita, dime por qué, Tu dolor hoy te enc...   
4688  [Angora fur, the Aegean Sea, It's a soft wet l...   
4689  [Angora fur, Aegean sea, It's a soft wet love ...   
4703  [I am a witness for the prosecution of a hate ...   
4704  [This is the story of two childhood sweetheart...   
4945  [Bebot bebot bet, Bebot bebot bet, Bebot bebot...   
5254  [Life's a movie, let the camera roll, Fast lif...   
5293  [Well sometimes I go out, by myself, and I loo...   
5321  [Jo, jo, vid Waterloo Napoleon fick ge sig, Me...   
5383  [Puedes escuchar Fernando?, Me recuerda tiempo...   
5384  [Yo por ti me engañe, hace tiempo lo sé, Y ya ...   
5385  [Chiquitita, dime por qué, Tu dolor hoy te enc...   
5395  [Varför sörjer du Fernando?, Varför klingar di...   
5546  [Oh oh oh oh, Oh oh oh oh, Ah!, Oh oh oh oh, A...   
5610  [It's going down, I'm yelling timber, You bett...   
6064  [Caution when it comes to love, I did, I did, ...   
6320  [It's been a long day without you, my friend, ...   
6888  [Dub, , Ponme el nombre que tú quieras, Nueva ...   

                                            song         artist  \
135                      lovestorytaylorsversion    taylorswift   
143                     thebestdaytaylorsversion    taylorswift   
148                      superstartaylorsversion    taylorswift   
149          theothersideofthedoortaylorsversion    taylorswift   
409   joannewheredoyouthinkyouregoinpianoversion       ladygaga   
632             aslongasyoulovemeacousticversion   justinbieber   
634          shedontlikethelightsacousticversion   justinbieber   
635                       takeyouacousticversion   justinbieber   
637             allaroundtheworldacousticversion   justinbieber   
1662                           ilovemeemoversion     demilovato   
1670                   reallydontcaresoloversion     demilovato   
1924                      playgroundfinalversion     lanadelrey   
1951                summertimethegershwinversion     lanadelrey   
1962                       trashmagicdemoversion     lanadelrey   
2722               bestsongevermusicvideoversion   onedirection   
2859             youmyeverythingalternateversion  elliegoulding   
2860          youmyeverythingostskinsfireversion  elliegoulding   
3168           heroesandvillainsalternateversion      beachboys   
4080                    chiquititaspanishversion           cher   
4688                      softandwetfunkyversion         prince   
4689           softandwetsecondunreleasedversion         prince   
4703              witness4theprosecutionversion1         prince   
4704              witness4theprosecutionversion2         prince   
4945                         bebotenglishversion  blackeyedpeas   
5254              youalreadyknowinterludeversion         fergie   
5293                            valerie68version   amywinehouse   
5321                      waterlooswedishversion        

In [52]:
pop_df = pop_df[~pop_df['song'].str.contains('version')]

Let's also remove songs that have "acoustic" in them...

In [53]:
pop_df[pop_df['song'].str.contains('acoustic')]

lyrics  \
1493  [Talkin' in my sleep at night, Makin' myself c...   
2157  [When I was six years old I broke my leg, I wa...   
2166  [My mind is a warrior,, My heart is a foreigne...   
2170  [I'm at a party I don't wanna be at, And I don...   
2184  [Mmmm, , A club isn't the best place to find a...   
2191  [She got the mm brown eyes, caramel thighs, lo...   
2386  [It's crazy when, The thing you love the most ...   
2604  [You lift my heart up when the rest of me is d...   
6263  [Ohh, Yeah, , I don't wanna hurt you baby, no ...   
6461  ['Cause after the afterparty, we're gonna keep...   
6554  [We should take this back to my place, That's ...   

                               song       artist  \
1493      newrulespianoacousticlive      dualipa   
2157        castleonthehillacoustic    edsheeran   
2166                 grade8acoustic    edsheeran   
2170              idontcareacoustic    edsheeran   
2184             shapeofyouacoustic    edsheeran   
2191       southoftheborderacoustic    edsheeran   
2386              graveyardacoustic       halsey   
2604                  latchacoustic     samsmith   
6263  idontwannahurtyoubabyacoustic  charlieputh   
6461     aftertheafterpartyacoustic    charlixcx   
6554              slowhandsacoustic   niallhoran   

                                          lyrics_string  
1493  Talkin' in my sleep at night,Makin' myself cra...  
2157  When I was six years old I broke my leg,I was ...  
2166  My mind is a warrior,,My heart is a foreigner,...  
2170  I'm at a party I don't wanna be at,And I don't...  
2184  Mmmm,,A club isn't the best place to find a lo...  
2191  She got the mm brown eyes, caramel thighs, lon...  
2386  It's crazy when,The thing you love the most is...  
2604  You lift my heart up when the rest of me is do...  
6263  Ohh,Yeah,,I don't wanna hurt you baby, no more...  
6461  'Cause after the afterparty, we're gonna keep ...  
6554  We should take this back to my place,That's wh...

In [54]:
pop_df = pop_df[~pop_df['song'].str.contains('acoustic')]

In [55]:
pop_df

lyrics  \
0     [He said the way my blue eyes shined, Put thos...   
1     [State the obvious, I didn't get my perfect fa...   
4     [You have a way of coming easily to me, And wh...   
7     [Cory's eyes are like a jungle, He smiles, it'...   
11    [Last Christmas I gave you my heart, But the v...   
...                                                 ...   
6961  [(I'll be home, I'll be home), , I'm dreaming ...   
6962  [Lose a layer or two and let's get lost in tim...   
6963  [Sorry I'm not so merry, But I feel like this ...   
6964  [Straight up, Tell me everything you've been t...   
6966  [I'm thinking of you, I'm thinking of you, I'm...   

                       song       artist  \
0                 timmcgraw  taylorswift   
1             picturetoburn  taylorswift   
4                 coldasyou  taylorswift   
7             staybeautiful  taylorswift   
11            lastchristmas  taylorswift   
...                     ...          ...   
6961  illbehomeforchristmas        mabel   
6962           knowmebetter        mabel   
6963    loneliesttimeofyear        mabel   
6964            myboymytown        mabel   
6966          thinkingofyou        mabel   

                                          lyrics_string  
0     He said the way my blue eyes shined,Put those ...  
1     State the obvious, I didn't get my perfect fan...  
4     You have a way of coming easily to me,And when...  
7     Cory's eyes are like a jungle,He smiles, it's ...  
11    Last Christmas I gave you my heart,But the ver...  
...                                                 ...  
6961  (I'll be home, I'll be home),,I'm dreaming ton...  
6962  Lose a layer or two and let's get lost in time...  
6963  Sorry I'm not so merry,But I feel like this ye...  
6964  Straight up,Tell me everything you've been thr...  
6966  I'm thinking of you,I'm thinking of you,I'm th...  

[5703 rows x 4 columns]

Now let's remove some offensive language. I have replaced the words with X.

In [59]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

In [60]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

In [ ]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

In [ ]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

In [ ]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

Finally, I can write the cleaned DataFrame to a csv file.

In [61]:
pop_df.to_csv('/Users/liamgentile/Desktop/pop_df.csv')

----------

# Rap Lyrics

In [62]:
rap_df = pd.read_csv('/Users/liamgentile/Desktop/lyric_files/rap_lyrics.csv', converters={'lyrics': eval})

In [63]:
rap_df

Unnamed: 0                                             lyrics  \
0               0  [[Intro:], Awww yeah! We in the motherfuckin' ...   
1               1  ["His opponent from St. Petersburg, Florida", ...   
2               2  [Hey yo you ready? Yeah I'm ready right, the f...   
3               3  [[Intro:], Bronsolino, Fuck that sitting-down ...   
4               4  [When I'm alone, Smoking weed, sitting by the ...   
...           ...                                                ...   
12364       12364  [This is revolution shit, Uh, yeah, c'mon and ...   
12365       12365  [It got the summer written all over it, It is ...   
12366       12366  [The meaning of God body is simple, It means y...   
12367       12367  [Rest in beats from Heavy D to Eazy E, The Not...   
12368       12368  [Tonight I can't sleep, I just stare at the wa...   

                         song         artist  
0                  moonstruck  actionbronson  
1               barryhorowitz  actionbronson  
2                  themadness  actionbronson  
3                 larrycsonka  actionbronson  
4               ronniecoleman  actionbronson  
...                       ...            ...  
12364  fightthepowerremix2020    publicenemy  
12365                  goatit    publicenemy  
12366        dontlookatthesky    publicenemy  
12367             restinbeats    publicenemy  
12368              ripblackat    publicenemy  

[12369 rows x 4 columns]

I'll start by removing the extra index column.

In [64]:
rap_df.drop('Unnamed: 0', axis=1, inplace=True)
rap_df

lyrics  \
0      [[Intro:], Awww yeah! We in the motherfuckin' ...   
1      ["His opponent from St. Petersburg, Florida", ...   
2      [Hey yo you ready? Yeah I'm ready right, the f...   
3      [[Intro:], Bronsolino, Fuck that sitting-down ...   
4      [When I'm alone, Smoking weed, sitting by the ...   
...                                                  ...   
12364  [This is revolution shit, Uh, yeah, c'mon and ...   
12365  [It got the summer written all over it, It is ...   
12366  [The meaning of God body is simple, It means y...   
12367  [Rest in beats from Heavy D to Eazy E, The Not...   
12368  [Tonight I can't sleep, I just stare at the wa...   

                         song         artist  
0                  moonstruck  actionbronson  
1               barryhorowitz  actionbronson  
2                  themadness  actionbronson  
3                 larrycsonka  actionbronson  
4               ronniecoleman  actionbronson  
...                       ...            ...  
12364  fightthepowerremix2020    publicenemy  
12365                  goatit    publicenemy  
12366        dontlookatthesky    publicenemy  
12367             restinbeats    publicenemy  
12368              ripblackat    publicenemy  

[12369 rows x 3 columns]

Now i'll create the lyrics_string column, which has been so helpful in cleaning the previous two DataFrames.

In [65]:
rap_df['lyrics_string'] = rap_df['lyrics'].apply(lambda x: ','.join(map(str, x)))
rap_df

lyrics  \
0      [[Intro:], Awww yeah! We in the motherfuckin' ...   
1      ["His opponent from St. Petersburg, Florida", ...   
2      [Hey yo you ready? Yeah I'm ready right, the f...   
3      [[Intro:], Bronsolino, Fuck that sitting-down ...   
4      [When I'm alone, Smoking weed, sitting by the ...   
...                                                  ...   
12364  [This is revolution shit, Uh, yeah, c'mon and ...   
12365  [It got the summer written all over it, It is ...   
12366  [The meaning of God body is simple, It means y...   
12367  [Rest in beats from Heavy D to Eazy E, The Not...   
12368  [Tonight I can't sleep, I just stare at the wa...   

                         song         artist  \
0                  moonstruck  actionbronson   
1               barryhorowitz  actionbronson   
2                  themadness  actionbronson   
3                 larrycsonka  actionbronson   
4               ronniecoleman  actionbronson   
...                       ...            ...   
12364  fightthepowerremix2020    publicenemy   
12365                  goatit    publicenemy   
12366        dontlookatthesky    publicenemy   
12367             restinbeats    publicenemy   
12368              ripblackat    publicenemy   

                                           lyrics_string  
0      [Intro:],Awww yeah! We in the motherfuckin' pl...  
1      "His opponent from St. Petersburg, Florida","T...  
2      Hey yo you ready? Yeah I'm ready right, the fa...  
3      [Intro:],Bronsolino,Fuck that sitting-down rap...  
4      When I'm alone,Smoking weed, sitting by the wi...  
...                                                  ...  
12364  This is revolution shit,Uh, yeah, c'mon and ge...  
12365  It got the summer written all over it,It is ti...  
12366  The meaning of God body is simple,It means you...  
12367  Rest in beats from Heavy D to Eazy E,The Notor...  
12368  Tonight I can't sleep, I just stare at the wal...  

[12369 rows x 4 columns]

I'll go ahead and remove all songs that contain the tag 'remix'...

In [66]:
rap_df[rap_df['song'].str.contains('remix')]

lyrics  \
152    [Say it..., , [Verse 1: Action Bronson], Well ...   
330    [[Styles P], Locked up, they won't let me out,...   
342    [Konvict, Konvict, Konvict, , Nobody wanna see...   
371    [I guess that I don't need that though, Now yo...   
372    [Money dup blow so ride 'round with that Nina ...   
...                                                  ...   
12117  [(Oh, is that Kaniel again?), Go crazy (Oh, ye...   
12131  [[Twice:], Why you wanna go away for the whole...   
12236  [Skills to kill, And fill a hole, we roll deep...   
12296  [[Paris], Take it back to the days when we rai...   
12364  [This is revolution shit, Uh, yeah, c'mon and ...   

                             song         artist  \
152         putitinyourmouthremix  actionbronson   
330                 lockedupremix           akon   
342        dontmattercalypsoremix           akon   
371               usedtoknowremix           akon   
372              cashinoutreremix           akon   
...                           ...            ...   
12117                gocrazyremix      youngthug   
12131      nowendyscontrollaremix      youngthug   
12236            livininazooremix    publicenemy   
12296  fieldniggaboogiexlr8rremix    publicenemy   
12364      fightthepowerremix2020    publicenemy   

                                           lyrics_string  
152    Say it...,,[Verse 1: Action Bronson],Well you ...  
330    [Styles P],Locked up, they won't let me out,,A...  
342    Konvict, Konvict, Konvict,,Nobody wanna see us...  
371    I guess that I don't need that though,Now you'...  
372    Money dup blow so ride 'round with that Nina (...  
...                                                  ...  
12117  (Oh, is that Kaniel again?),Go crazy (Oh, yeah...  
12131  [Twice:],Why you wanna go away for the whole s...  
12236  Skills to kill,And fill a hole, we roll deep,W...  
12296  [Paris],Take it back to the days when we raise...  
12364  This is revolution shit,Uh, yeah, c'mon and ge...  

[280 rows x 4 columns]

In [67]:
rap_df = rap_df[~rap_df['song'].str.contains('remix')]

Now i'll remove songs where the song name contains 'version'...

In [68]:
rap_df[rap_df['song'].str.contains('version')]

lyrics  \
316    [[Intro:], , Just some shit, , [Verse:], Three...   
1712   [[Part 1: Faithful], , You expensive, you know...   
2126   [Arm stretch a tee like I nailed it, Raf movin...   
4723   [[Intro: Swizz Beats], Swizzy...Beats..., , [C...   
5234   [[Kendrick Lamar:], Love, let's talk about lov...   
6373   [[Hook:], All the girls make your heels clink ...   
6662   [Oh damn, Goddamn Based God, you got bitches, ...   
8924   [1,2 Uhh 1,1, , M-E-T, H-O-D, MAN, M-E-T, H-O-...   
10651  [[Intro: P.R. Terrorist, (RZA), {Killa Sin}], ...   
11411  [Yeah, Fargo, , Shawty, you know how I'm strok...   

                                     song         artist  \
316                  untouchablearversion      aesoprock   
1712              faithfulextendedversion          drake   
2126                    bikingsoloversion     frankocean   
4723                 stopalternateversion           jayz   
5234             allthestarssingleversion  kendricklamar   
6373               clinkclinkcleanversion           lilb   
6662   prettyboychoppedandscrewedversion2           lilb   
8924            methodmanhomegrownversion      methodman   
10651            terroristjapaneseversion            rza   
11411           bodmonsongextendedversion      torylanez   

                                           lyrics_string  
316    [Intro:],,Just some shit,,[Verse:],Three's com...  
1712   [Part 1: Faithful],,You expensive, you know th...  
2126   Arm stretch a tee like I nailed it,Raf movin' ...  
4723   [Intro: Swizz Beats],Swizzy...Beats...,,[Choru...  
5234   [Kendrick Lamar:],Love, let's talk about love,...  
6373   [Hook:],All the girls make your heels clink cl...  
6662   Oh damn, Goddamn Based God, you got bitches,Oh...  
8924   1,2 Uhh 1,1,,M-E-T, H-O-D, MAN,M-E-T, H-O-D, M...  
10651  [Intro: P.R. Terrorist, (RZA), {Killa Sin}],Bo...  
11411  Yeah, Fargo,,Shawty, you know how I'm strokin'...

In [69]:
rap_df = rap_df[~rap_df['song'].str.contains('version')]

Now let's investigate songs that contain square brackets (non-lyrical content) at some point...

In [70]:
rap_df[rap_df['lyrics_string'].str.contains(']')]

lyrics  \
0      [[Intro:], Awww yeah! We in the motherfuckin' ...   
1      ["His opponent from St. Petersburg, Florida", ...   
2      [Hey yo you ready? Yeah I'm ready right, the f...   
3      [[Intro:], Bronsolino, Fuck that sitting-down ...   
4      [When I'm alone, Smoking weed, sitting by the ...   
...                                                  ...   
12338  [[James Bomb:], I boarded a plane headstrong, ...   
12339  [[Intro - Chuck D, Flavor Flav], Lost in space...   
12340  [(I bite the apple, And the apple bite me back...   
12341  [[Verse 1 - Chuck D:], A - The war of art agai...   
12355  [I am, Louder than fuck, Mercy wanna curse me,...   

                               song         artist  \
0                        moonstruck  actionbronson   
1                     barryhorowitz  actionbronson   
2                        themadness  actionbronson   
3                       larrycsonka  actionbronson   
4                     ronniecoleman  actionbronson   
...                             ...            ...   
12338                     mineagain    publicenemy   
12339              lostinspacemusic    publicenemy   
12340            corplantationopoly    publicenemy   
12341                     earthizen    publicenemy   
12355  stdslaverytransmitteddisease    publicenemy   

                                           lyrics_string  
0      [Intro:],Awww yeah! We in the motherfuckin' pl...  
1      "His opponent from St. Petersburg, Florida","T...  
2      Hey yo you ready? Yeah I'm ready right, the fa...  
3      [Intro:],Bronsolino,Fuck that sitting-down rap...  
4      When I'm alone,Smoking weed, sitting by the wi...  
...                                                  ...  
12338  [James Bomb:],I boarded a plane headstrong,Lan...  
12339  [Intro - Chuck D, Flavor Flav],Lost in space.....  
12340  (I bite the apple,And the apple bite me back),...  
12341  [Verse 1 - Chuck D:],A - The war of art agains...  
12355  I am,Louder than fuck,Mercy wanna curse me,Don...  

[7381 rows x 4 columns]

7381 songs! Perhaps this is to be expected as often there is more than one rapper on a song, and so they have a tag to indicate who is rapping each verse. Fortunately I have each of my songs as a list of lines. Ill just remove those items in square brackets by employing a nested for loop on the lyrics column.

In [71]:
for array in rap_df['lyrics']:
    for x in array:
        if ']' in x:
            array.remove(x)

Finally, I'll go ahead and remove lines that contain potentially offensive language. As mentioned above, I have replaced the words with X.

In [ ]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

In [ ]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

In [ ]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

In [ ]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

In [ ]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

In [ ]:
for array in pop_df['lyrics']:
    for x in array:
        if 'X' in x:
            array.remove(x)

Finally, I will save the cleaned DataFrame to csv.

In [75]:
rap_df.to_csv('/Users/liamgentile/Desktop/rap_df.csv')